src/h5web/providers/my-utils.ts

```diff
@@ -149,7 +148,10 @@ export function makeMyNxDataGroup<
     errors?: MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>;
     title?: MyHDF5Dataset<HDF5ScalarShape, HDF5StringType>;
     silxStyle?: SilxStyle;
-  } & AllOrNone<{ axes: T; axesAttr: (keyof T | '.')[] }> &
+  } & (
+    | { axes: T; axesAttr: (Extract<keyof T, string> | '.')[] }
+    | { axes?: never; axesAttr?: never }
+  ) &
```

Comment:
Changing my mind on this. I think inlining is clearer after all. 🤷‍♂️ 

The change below also required a slight change in typing: `Extract<keyof T, string>` instead of `keyof T`, as `keyof T` resolves to `string | number | Symbol` as explained in https://github.com/microsoft/TypeScript/issues/23724#issuecomment-384807714 ... even though `T` is `Record<string, ...>`! This seems very odd... there's a comment about it in the linked PR, but no answer.

Reply:
Probably `keyof` does not behave well with templates ? :shrug:

Reply:
Generics you mean? The guy's comment on https://github.com/microsoft/TypeScript/issues/23724#issuecomment-384807714 doesn't mention a generic so I don't think it's related.

Reply:
Link to the comment: https://github.com/microsoft/TypeScript/issues/23724#issuecomment-638840419